### PR TITLE
Actions Refactoring

### PR DESCRIPTION
**Pull request by jifeon**
- actions are separated into the heap of files
- you can now build browser bundle just with npm run build, the problem with moment is fixed

### CODE REVIEW

File: lib/Actions.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
 
-var debug = require('debug')('Actions'),
-    vow = require('vow'),
-    _ = require('lodash'),
-    Storage = require('./Storage');
+const debug = require('debug')('Actions');
+const vow = require('vow');
+const _ = require('lodash');
+const Storage = require('./Storage');
+const actionsFactory = require('./actions/actionsFactory');
```
Comment (jifeon) [inline lib/Actions.js]: using actionsFactory to instantiate actions now, Actions.js is reduced by 300+ lines

File: webpack.config.js [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
-                //query: {
-                    //presets: ['es2015']
-                //}
+                loader: 'babel'
             }
         ]
-    }
+    },
+    plugins: [
+        new webpack.ContextReplacementPlugin(/moment[\/\\]locale$/, /en|ru/)
```
Comment (jifeon) [inline webpack.config.js]: now moment is built with en and ru locales only, so we can just run `npm run build` before commit without changing `Tranform.js` 🐗

File: lib/Actions.js (hunk 1 above)
Comment (maZahaca) [inline lib/Actions.js]: Why is it small letter at the begging of the actionsFactory?

File: lib/Actions.js [hunk 3]
```diff
@@ -313,99 +223,22 @@ Actions.prototype = {
...
             return promises[wonCase];
         });
     },
-    
-    /**
-     * Add custom action
-     * @param {string} type
-     * @param {Function} action
-     */
-    addAction: function (type, action) {
```
Comment (maZahaca) [inline lib/Actions.js]: Don't we have ability to extend actions anymore?

File: lib/Parser.js [hunk 4]
```diff
@@ -194,7 +195,7 @@ Parser.prototype = {
      * @see {@link Actions#addAction}
      */
     addAction: function() {
-        return this._actions.addAction.apply(this._actions, arguments)
+        return actionsFactory.addAction.apply(actionsFactory, arguments)
```
Comment (maZahaca) [inline lib/Parser.js]: Okay, found an answer, cut previous question

File: lib/actions/ActionClick.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+            .evaluateJs(this._selector, /* @covignore */ function(selector) {
+                var nodes = Sizzle(selector);
+                for (var i = 0, l = nodes.length; i < l; i++) {
+                    nodes[i].click();
+                }
+
+                return nodes.length;
+            })
+
```
Comment (maZahaca) [inline lib/actions/ActionClick.js]: Why those break lines are needed?

File: lib/actions/ActionOpen.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+ * Opens a page
+ */
+
+'use strict';
+
+const Action = require('./Action');
+
+class ActionOpen extends Action {
+    perform () {
+        return this._env.evaluateJs(this._options.url, function (url) {
```
Comment (maZahaca) [inline lib/actions/ActionOpen.js]: Add please 'covignore' before the function

File: lib/Actions.js (hunk 1 above)
Comment (jifeon) [inline lib/Actions.js]: 'cause it's not a class, it's a singleton, usual object.

File: lib/actions/ActionClick.js (hunk 5 above)
Comment (jifeon) [inline lib/actions/ActionClick.js]: fixed

File: lib/actions/ActionOpen.js (hunk 6 above)
Comment (jifeon) [inline lib/actions/ActionOpen.js]: fixed

**Pull request merged**